shared/message/input/PS3Controller.proto

```diff
@@ -0,0 +1,246 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2013 NUbots <nubots@nubots.net>
+ */
+
+syntax = "proto3";
+
+package message.input.PS3Controller;
```

Comment:
Packages don’t have things like this! This should probably either be snake case, or make PS3Controller a message.

Reply:
what?

Reply:
You've made a namespace `message::input::PS3Controller`, the last component looks like a class but it's a namespace. You should either make it a class (message) or make it conform to the namespace (make it `ps3controller` over `PS3Controller`